### ISSUE TITLE
Database Structure

### ISSUE THREAD

**Issue by cursemeslowly** [cursemeslowly/devrantia#2]
Now that we have at least better understanding of this project. What are your thoughts on database structure? Below is what I have done just now. So that we have a starting point to discuss and improve.

![DB Structure Sketch 0.1](https://cursemeslowly.github.io/devrantia/uploads/db-structure-sketch-0.1.jpg)

**Comment by cursemeslowly** [comment]
Need to consider how we gonna structure for followings as well

- map/locations
- awards
- npcs

**Comment by Stuni907** [comment]
Do you think you will support multi language?
What do you mean with roles?

**Comment by cursemeslowly** [comment]
@Stuni907 What do you suggest with the multi language? Should we start with english only first and implement it later or prepare for multi from the start?

Roles as in we gonna have at least admins and users/players, I guess. 🤔 
Roles will have different permissions.

For the player level role, it should be in other table.

**Comment by Stuni907** [comment]
@cursemeslowly 
Example: There is a table task_types with columns _name_ and _langCode_.
Default it's English (1) and if you want to add a language you add a code.

**Comment by Derwaan** [comment]
@cursemeslowly 
For the roles of the user, if it is based on an admin/user system, why not include a columns role in the user model?
Or it is based on a permission system like read/write access?

Besides, maybe using UML for representing the database if the complexity continues to increase :thinking: I don't know about any collaborative UML tool, but that's probably a thing.

**Comment by cursemeslowly** [comment]
@Derwaan yeah. I also think about that. I know some collab mind-map app. Not sure if we can use for our purpose or not.

For the roles, yeah I guess yes we can go as simple as a column for role in users table too.
I am just not very sure if we gonna need new roles in future or not. May be I'm over thinking.

**Comment by cursemeslowly** [comment]
So this is what comes next.

![img_20180202_150803](https://user-images.githubusercontent.com/35447763/35724279-8933dd98-082b-11e8-8aba-7b96436186bb.jpg)
